juju/model.py

```diff
@@ -1398,7 +1398,8 @@ async def _notify_observers(self, delta, old_obj, new_obj):
 
         for o in self._observers:
             if o.cares_about(delta):
-                asyncio.ensure_future(o(delta, old_obj, new_obj, self))
+                f = asyncio.ensure_future(o(delta, old_obj, new_obj, self))
+                f.add_done_callback(lambda f: f.result())
```

Comment:
I'd rather this hack was not removed.
It's most likely there to prevent "Task was never awaited for" warnings.
Not a great practice, but out of scope of this PR.